src/base/static/components/organisms/main-map.js

```diff
@@ -313,19 +314,18 @@ class MainMap extends Component {
     }
 
     if (this.props.placeFilters.length !== prevProps.placeFilters.length) {
-      // Filters have been applied or unapplied.
-
-      // "sourceId" and "datasetSlug" are the same for Place sources. We assume
-      // that all filters will be filtering the same source, so it's safe to
-      // pull the datasetSlug/sourceId from the first filter in the filters
-      // array.
-      const sourceId = this.props.placeFilters.length
-        ? this.props.placeFilters[0].datasetSlug
-        : prevProps.placeFilters[0].datasetSlug;
-      this.props.updateFeaturesInGeoJSONSource(
-        sourceId,
-        createGeoJSONFromPlaces(this.props.filteredPlaces).features,
-      );
+      // Filters have been added or removed.
+
+      this.props.datasets.map(dataset => dataset.slug).forEach(sourceId => {
+        this.props.updateFeaturesInGeoJSONSource(
+          sourceId,
+          createGeoJSONFromPlaces(
+            this.props.filteredPlaces.filter(
+              place => place._datasetSlug === sourceId,
+            ),
+          ).features,
+        );
+      });
```

Comment:
Calculating geojson and firing these action on each dataset is not performant, but then again, filters are secondary/edge case features, so I'm okay with this for now.

Longer term, we can consider indexing places by dataset id's.